kroxylicious-kms-test-support/src/main/java/io/kroxylicious/kms/service/RecordEncryptionTestInvocationContextProvider.java

```diff
@@ -53,14 +50,14 @@ public List<Extension> getAdditionalExtensions() {
             }
 
             return List.of(
-                    (BeforeTestExecutionCallback) extensionContext -> kmsFacade.start(),
+                    (BeforeEachCallback) extensionContext -> kmsFacade.start(),
                     new TypeBasedParameterResolver<TestKmsFacade<?, ?, ?>>() {
                         @Override
                         public TestKmsFacade<?, ?, ?> resolveParameter(ParameterContext parameterContext, ExtensionContext extensionContext) {
                             return kmsFacade;
                         }
                     },
-                    (AfterTestExecutionCallback) extensionContext -> kmsFacade.stop());
+                    (AfterEachCallback) extensionContext -> kmsFacade.stop());
```

Comment:
This increases test isolation, but at the expense of being slower I guess. Do you have any sense of the impact? If cheap enough I prefer the isolation.

Reply:
I don't _think_ it adds runtime.  My motivation for change was BeforeEachCallback ran too late (after KmsIT#beforeEach).  I needed the execution order to be the other way about so that the KmsService gets wired to the KMS backend just stood up by the test.

Reply:
Hmm maybe I'm wrong I thought it was moving from calling `kms.start` once per class to once per test. If I'm wrong on that then great!